common/src/main/java/jp/co/soramitsu/common/list/PayloadGenerator.kt

```diff
@@ -12,18 +12,26 @@ open class PayloadGenerator<T>(private vararg val checks: DiffCheck<T, *>) {
     }
 }
 
+typealias UnknownPayloadHandler = (Any?) -> Unit
+
 @Suppress("UNCHECKED_CAST")
 fun <T, VH : RecyclerView.ViewHolder> ListAdapter<T, VH>.resolvePayload(
     holder: VH,
     position: Int,
     payloads: List<Any>,
-    onDiffCheck: (DiffCheck<T, *>) -> Unit
+    onUnknownPayload: UnknownPayloadHandler? = null,
+    onDiffCheck: (DiffCheck<T, *>) -> Unit,
 ) {
     if (payloads.isEmpty()) {
         onBindViewHolder(holder, position)
     } else {
-        val diffChecks = payloads.first() as List<DiffCheck<T, *>>
+        when (val payload = payloads.first()) {
+            is List<*> -> {
```

Comment:
I propose to write just 
` when (val payload = payloads.first()) {
             is List<DiffCheck<T, *>> -> { `

Reply:
Due to type erasure you cannot check for `List<T>`, only for `List<*>`